src/GafferScene/RenderController.cpp

```diff
@@ -1614,10 +1621,15 @@ void RenderController::updateInternal( const ProgressCallback &callback, const I
 
 		if( m_dirtyGlobalComponents & SetsGlobalComponent )
 		{
-			if( m_renderSets.update( m_scene.get() ) & ( Private::RendererAlgo::RenderSets::RenderSetsChanged | Private::RendererAlgo::RenderSets::SoloLightsSetChanged ) )
+			unsigned int changedSets = m_renderSets.update( m_scene.get() );
+			if( changedSets & Private::RendererAlgo::RenderSets::RenderSetsChanged )
 			{
 				m_changedGlobalComponents |= RenderSetsGlobalComponent;
 			}
+			if( changedSets & Private::RendererAlgo::RenderSets::SoloLightsSetChanged )
+			{
+				m_changedGlobalComponents |= SoloLightsGlobalComponent;
+			}
```

Comment:
I don't understand what this commit is trying to achieve - see my comments about `RenderSets::SoloLightsSetChanged`. The state tracking for RenderController is tricky, so the simpler we can keep it the better.

Reply:
I was a bit turned around on how exactly the updates were being used. I've dropped this commit and tucked both updates under the `AttributesChanged` as you mentioned above. That's in 7039b7fe28f34070523dfba4aac889e7cd18a62f